File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        map<int,int> maps {};
        for(const auto& num: nums) {
            if(maps.contains(num)) {
                if(maps[num] == 2) {
                    maps.erase(num);
                } else {
                    maps[num]=2;
                }
            }
            else {
                maps[num] = 1;
            }
        }

        return maps.begin()->first;
    }
};

File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        numDict = {}
        for num in nums:
            if num in numDict:
                if numDict[num] == 2:
                    del numDict[num]
                else:
                    numDict[num]=2
            else:
                numDict[num]=1
        return list(numDict.keys())[0]
                    
        